File: spring-rest2ts-examples/web/src/app/_model/example/biz/core/ITableObj.ts
//此代码为生成的代码请勿修改 !!!
import {DetailObj} from '../../../example/biz/core/DetailObj'
import {InsertObj} from '../../../example/biz/core/InsertObj'
import {QueryObj} from '../../../example/biz/core/QueryObj'
import {UpdateObj} from '../../../example/biz/core/UpdateObj'
import {Observable} from 'rxjs';

export interface ITableObj <I extends any, U extends any, Q extends any, D extends any> extends  DetailObj<D> , InsertObj<I> , QueryObj<Q> , UpdateObj<U> {

delete(keyIDs: number[]): Observable<void> ;

}

File: spring-rest2ts-examples/java/src/main/java/example/biz/core/ITableObj.java
package example.biz.core;


import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * @param <Q> query 查询一条数据的泛型 model类
 * @param <I> insert 新增一条数据的泛型 model类
 * @param <U> update 修改一条数据的泛型 model类
 * @param <D> detail 查看一天数据详情的model类
 */
@Controller
public interface ITableObj<I,U,Q,D> extends InsertObj<I>,UpdateObj<U>,QueryObj<Q>,DetailObj<D>  {



    /**
     * 删除表对象
     *
     * @param keyIDs
     * @throws Exception
     */
    @RequestMapping(value = "delete")
    @ResponseBody
    void delete(@RequestBody List<Long> keyIDs) throws Exception;
}
